compiler/rustc_passes/src/check_attr.rs

```diff
@@ -561,12 +561,15 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
         allowed_target: Target,
     ) {
         if target != allowed_target {
+            let path = attr.path();
+            let path: Vec<_> = path.iter().map(|s| s.as_str()).collect();
+            let attr_name = path.join("::");
```

Comment:
if this happens a lot throughout the compiler it's a bit ugly to keep iterating and joining these. In HIR I think I had a solution for this, with hir::AttributePath which implements Display. I'll add this to my own TODO list, to make this nicer at the same time when AttributeExt is removed :)